[production][package#show] Symbol or String expected, but NilClass given.

package#show threw this 3 times
<pre>
Symbol or String expected, but NilClass given.
</pre>
[https://build.opensuse.org/package/show/devel:gcc/gcc33](https://build.opensuse.org/package/show/devel:gcc/gcc33)"


[See this exception on Errbit](https://errbit-opensuse.herokuapp.com/apps/5620ca2bdc71fa00b1000000/problems/5d5f9c259df855000dda726e)

executive summary:

coderay crashes rendering code in markdown comments:

```
<comments project="devel:gcc" package="gcc33">
  <comment who="badshah400" when="2018-12-07 12:52:46 UTC" id="812733">@rguenther Any suggestions on how to fix builds on openSUSE 15.0 and Factory (last successful build for Factory is from more than a year old)? Thanks!&#13;
&#13;
```&#13;
make[4]: Entering directory '/home/abuild/rpmbuild/BUILD/gcc-3.3.3/obj-x86_64-suse-linux/gcc'&#13;
[  112s] ../../gcc/libgcc2.c: In function `gcov_exit':&#13;
[  112s] ../../gcc/libgcc2.c:1307: warning: unused variable `program_max_sum'&#13;
[  112s] ../../gcc/libgcc2.c:1309: warning: unused variable `program_runs'&#13;
[  112s] ../../gcc/unwind-dw2.c: In function `uw_frame_state_for':&#13;
[  112s] ../../gcc/unwind-dw2.c:953: error: dereferencing pointer to incomplete type&#13;
[  112s] ../../gcc/unwind-dw2.c: In function `amd64_frame_state_for':&#13;
[  112s] ../../gcc/unwind-dw2.c:1313: error: dereferencing pointer to incomplete type&#13;
```</comment>
</comments>

--- a/src/api/lib/obsapi/markdown_renderer.rb
+++ b/src/api/lib/obsapi/markdown_renderer.rb
@@ -1,4 +1,5 @@
 require 'uri'
+require 'cgi'
 
 module OBSApi
   class MarkdownRenderer < Redcarpet::Render::HTML
@@ -35,10 +36,11 @@ module OBSApi
         link = URI.join(::Configuration.obs_url, link)
       rescue URI::InvalidURIError
       end
-      "<a href='#{link}'#{title}>#{content}</a>"
+      "<a href='#{link}'#{title}>#{CGI.escape_html(content)}</a>"
     end
 
     def block_code(code, language)
+      language ||= :plaintext
       CodeRay.scan(code, language).div(css: :class)
     end
   end

--- a/src/api/spec/helpers/webui/markdown_helper_spec.rb
+++ b/src/api/spec/helpers/webui/markdown_helper_spec.rb
@@ -28,5 +28,31 @@ please review.</p>\n"
         "<p>anbox<a href='the number'>400000+22d000</a></p>\n"
       )
     end
+
+    it 'does not crash due to a missing language in a code block' do
+      expect(render_as_markdown("```\ntext\n```")).to eq("<div class=\"CodeRay\">\n  <div class=\"code\"><pre>text\n</pre></div>\n</div>\n")
+    end
+
+    it 'does apply a class to a code block with a language' do
+      expect(render_as_markdown("```ruby\ndef\n```")).to eq("<div class=\"CodeRay\">\n  <div class=\"code\"><pre><span class=\"keyword\">def</span>\n</pre></div>\n</div>\n")
+    end
+
+    it 'does remove dangerous html from the view' do
+      expect(render_as_markdown('<script></script>')).to eq("\n")
+    end
+
+    it 'does remove dangerous html from inside the code blocks with a language' do
+      expect(render_as_markdown("```html\n<script></script>\n```")).to eq(
+        "<div class=\"CodeRay\">\n  <div class=\"code\"><pre><span class=\"tag\">&lt;script&gt;</span><span class=\"tag\">&lt;/script&gt;</span>\n</pre></div>\n</div>\n"
+      )
+    end
+
+    it 'does remove dangerous html from inside the code blocks without a language' do
+      expect(render_as_markdown("```\n<script></script>\n```")).to eq("<div class=\"CodeRay\">\n  <div class=\"code\"><pre>&lt;script&gt;&lt;/script&gt;\n</pre></div>\n</div>\n")
+    end
+
+    it 'does remove dangerous html from inside the links' do
+      expect(render_as_markdown('[<script></script>](https://build.opensuse.org)')).to eq("<p><a href='https://build.opensuse.org'>&lt;script&gt;&lt;/script&gt;</a></p>\n")
+    end
   end
 end

